packages/block-library/src/block/edit.js

```diff
@@ -47,14 +57,37 @@ export default function ReusableBlockEdit( { attributes: { ref }, clientId } ) {
 				'wp_block',
 				ref,
 			] );
+			const blockHasEdits = select( coreStore ).hasEditsForEntityRecord(
+				'postType',
+				'wp_block',
+				ref
+			);
 			return {
 				hasResolved: hasResolvedBlock,
 				isMissing: hasResolvedBlock && ! persistedBlock,
+				hasEdits: blockHasEdits,
 			};
 		},
 		[ ref, clientId ]
 	);
 
+	const { isChildSelected, isSelected } = useSelect( ( select ) => {
+		const { getBlockParents, getSelectedBlockClientId } = select(
+			blockEditorStore
+		);
+		const selectedBlockClientId = getSelectedBlockClientId();
+		const _isSelected = selectedBlockClientId === clientId;
+
+		const parents = getBlockParents( selectedBlockClientId );
+		const firstParentClientId = parents[ parents.length - 1 ];
+		const _isChildSelected = firstParentClientId === clientId;
+
+		return {
+			isChildSelected: _isChildSelected,
+			isSelected: _isSelected,
+		};
+	}, [] );
```

Comment:
`clientId` is a dependency here.

Reply:
Thank you! You saved me from future errors! 🚀